plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2240,6 +2240,16 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         HiveTableHandle handle = (HiveTableHandle) tableHandle;
         checkArgument(handle.getAnalyzePartitionValues().isEmpty() || constraint.getSummary().isAll(), "Analyze should not have a constraint");
 
+        if (!canPredicateFilterPartitions(constraint) && !constraint.getSummary().isNone() && handle.getPartitions().isEmpty()) {
+            // Find a partition column that is constraint more strictly than with `IS NOT NULL` predicate
+            if (constraint.getSummary().getDomains().orElseThrow().entrySet().stream()
+                    .noneMatch(entry -> ((HiveColumnHandle) entry.getKey()).isPartitionKey() && !entry.getValue().getValues().isAll())) {
```

Comment:
nit: can we use filter(entry -> ((HiveColumnHandle) entry.getKey()).isPartitionKey()).noneMatch

Reply:
i thought about that